sql/distinct.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Tamir Duberstein (tamird@gmail.com)
+
+package sql
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/sql/parser"
+)
+
+// distinct constructs a distinctNode.
+func (*planner) distinct(n *parser.Select) (*distinctNode, error) {
+	distinct := distinctNode{}
+	if n.Distinct {
+		distinct.seen = make(map[string]struct{})
+	}
+	return &distinct, nil
```

Comment:
I think this should work like `planner.sort()` where we `return nil, nil` if `!n.Distinct`. Then in `distinctNode.wrap()` you do an initial check to see if `node == nil`.